dlt/normalize/normalize.py

```diff
@@ -139,31 +140,38 @@ def _w_normalize_chunk(load_storage: LoadStorage, schema: Schema, load_id: str,
                 # filter row, may eliminate some or all fields
                 row = schema.filter_row(table_name, row)
                 # do not process empty rows
-                if row:
-                    # decode pua types
-                    for k, v in row.items():
-                        row[k] = custom_pua_decode(v)  # type: ignore
-                    # coerce row of values into schema table, generating partial table with new columns if any
-                    row, partial_table = schema.coerce_row(table_name, parent_table, row)
-                    # theres a new table or new columns in existing table
-                    if partial_table:
-                        # update schema and save the change
-                        schema.update_schema(partial_table)
-                        table_updates = schema_update.setdefault(table_name, [])
-                        table_updates.append(partial_table)
-                        # update our columns
-                        column_schemas[table_name] = schema.get_table_columns(table_name)
-                    # get current columns schema
-                    columns = column_schemas.get(table_name)
-                    if not columns:
-                        columns = schema.get_table_columns(table_name)
-                        column_schemas[table_name] = columns
-                    # store row
-                    # TODO: it is possible to write to single file from many processes using this: https://gitlab.com/warsaw/flufl.lock
-                    load_storage.write_data_item(load_id, schema_name, table_name, row, columns)
-                    # count total items
-                    items_count += 1
-                    increase_row_count(row_counts, table_name, 1)
+                if not row:
+                    continue
+                # decode pua types
+                for k, v in row.items():
+                    row[k] = custom_pua_decode(v)  # type: ignore
+                # coerce row of values into schema table, generating partial table with new columns if any
+                row, partial_table = schema.coerce_row(table_name, parent_table, row)
+                # if we detect a migration, the check update
+                if partial_table:
+                    row, partial_table = schema.check_schema_update(parent_table, table_name, row, partial_table, schema_evolution_settings)
```

Comment:
ok this will not happen often. so we do not need to cache. but evolving new tables with columns frozen will surely fail. we can add just one column this way. we need to let add columns until normalizer finishes. and we need to test it

Reply:
Could you explain how a test would look exactly? As mentioned above, I am testing the creation of a table from a new resource and this works even if new columns are frozen.

Reply:
I just explained how to test that above